### PR TITLE
feat: add range filter when importing GitHub repos for an org

### PR DESCRIPTION
**Pull request by brandonroberts**
## Description

- Adds a day range filter when importing repos for a GitHub org
- Filters the imported repo list by the range based on the repo `pushed_at` date
- Also fixes an issue with importing large orgs like `microsoft`

## Related Tickets & Documents

Closes https://github.com/open-sauced/app/issues/3607

## Mobile & Desktop Screenshots/Recordings

image

## Steps to QA

1.  Login
2. Go to settings for a workspace
3. Click Add Repositories
4. Click on Import GitHub Organization
5. Select an Org
6. Note the repos is filtered to the date range

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [x] Tier 3
- [ ] Tier 4

### CODE REVIEW

**Review by zeucapua — APPROVED** [review]

File: components/Workspaces/TrackedRepoWizard/SelectOrgReposStep.tsx [hunk 1]
```diff
@@ -78,11 +83,19 @@ export const SelectOrgReposStep = ({
...
+          <span className="flex align-middle">{organization}</span>
+        </div>
+        <div className="flex justify-end">
+          <DayRangePicker onDayRangeChanged={(e) => setRange(Number(e))} />
+        </div>
       </div>
-      <p>Select the organization repositories that you want to track.</p>
+      <p>
+        Select the organization repositories updated in the last <span className="font-bold">{range} days</span> that
+        you want to track.
```
Comment (jpmcb) [inline components/Workspaces/TrackedRepoWizard/SelectOrgReposStep.tsx:97]: Thinking of an edge case here: is there now no way to import all repos from an org that go beyond being updated in 1 year? I.e., all repos regardless of a filter. For example, with the OpenSauced org, we end up missing all those old repos that we haven't touched in some time.

Which, if this is the intend, then that makes sense 👍🏼

File: components/Workspaces/TrackedRepoWizard/SelectOrgReposStep.tsx [hunk 2]
```diff
@@ -78,11 +83,19 @@ export const SelectOrgReposStep = ({
...
-        <Avatar contributor={organization} size="xsmall" />
-        <span>{organization}</span>
+      <div className="flex w-full justify-between gap-2 font-semibold">
+        <div className="flex gap-2 justify-start align-middle">
+          <Avatar contributor={organization} size="small" />
+          <span className="flex align-middle">{organization}</span>
+        </div>
+        <div className="flex justify-end">
+          <DayRangePicker onDayRangeChanged={(e) => setRange(Number(e))} />
+        </div>
```
Comment (nickytonline) [inline components/Workspaces/TrackedRepoWizard/SelectOrgReposStep.tsx:93]: refactor: no need for the div wrapper.

```suggestion
        <DayRangePicker onDayRangeChanged={(e) => setRange(Number(e))} />
```

**Review by nickytonline — APPROVED** [review]
One small tweak then good to 🚢

File: components/Workspaces/TrackedRepoWizard/SelectOrgReposStep.tsx (hunk 1 above)
Comment (brandonroberts) [reply components/Workspaces/TrackedRepoWizard/SelectOrgReposStep.tsx:97]: Yes, that's correct. Feedback given to @bdougie was that older repos that haven't been updated shouldn't be listed on the org import. I suppose they could go back and copy/paste the other ones in

File: components/Workspaces/TrackedRepoWizard/SelectOrgReposStep.tsx (hunk 2 above)
Comment (brandonroberts) [reply components/Workspaces/TrackedRepoWizard/SelectOrgReposStep.tsx]: Thanks!

**Pull request merged**